airbyte-workers/src/main/java/io/airbyte/workers/config/ApiClientBeanFactory.java

```diff
@@ -57,6 +58,29 @@ public AirbyteApiClient airbyteApiClient(
             }));
   }
 
+  @Singleton
+  public SourceApi sourceApi(
```

Comment:
I think you should be able to reuse the airbyteApiClient from above here:
```
@Singleton
public SourceApi sourceApi(final AirbyteApiClient airbytceApiClient) {
    return new SourceApi(airbyteApiClient);
}
```

Reply:
Got it. In this case, we should extract the the `ApiClient` into its own singleton, and inject it in both `AirbyteApiClient` and `SourceApi`. We can pair quickly to speed up the process if you want.

`ApiClient` is the client used to make the actual calls. It should come from the API code generator we use.

`AirbyteApiClient` is a class that currently consolidate all the Apis within one. The usage of this one should decrease, moving forward, we should inject the specific Apis to highlight dependencies rather than passing a catch all with everything we have.

Reply:
just updated